K2Bridge/Factories/BucketFactory.cs

```diff
@@ -38,51 +39,104 @@ public static DateHistogramBucket CreateDateHistogramBucketFromDataRow(DataRow r
                 Aggs = new System.Collections.Generic.Dictionary<string, System.Collections.Generic.List<double>>(),
             };
 
-            var clmns = row.Table.Columns;
-            foreach (DataColumn clmn in clmns)
+            CreateAggregationColumns(dhb, row);
+
+            return dhb;
+        }
+
+        /// <summary>
+        /// Create a new <see cref="Bucket" from a given <see cref="DataRow"/>/>.
+        /// </summary>
+        /// <param name="row">The row to be transformed to bucket.</param>
+        /// <returns>A new Bucket.</returns>
+        public static Bucket CreateTermsBucketFromDataRow(DataRow row)
+        {
+            Ensure.IsNotNull(row, nameof(row));
+
+            var key = row[(int)BucketColumnNames.SummarizeByColumn];
+            var count = row[BucketColumnNames.Count];
+
+            var tb = new Bucket
             {
-                if (clmn.ColumnName == BucketColumnNames.Count || clmns.IndexOf(clmn) == (int)BucketColumnNames.Timestamp)
-                {
-                    continue;
-                }
+                DocCount = Convert.ToInt32(count),
+                Key = Convert.ToString(key),
+                Aggs = new System.Collections.Generic.Dictionary<string, System.Collections.Generic.List<double>>(),
+            };
 
-                dhb.Aggs[clmn.ColumnName] = new System.Collections.Generic.List<double>() { Convert.ToDouble(row[clmn.ColumnName]) };
-            }
+            CreateAggregationColumns(tb, row);
 
-            return dhb;
+            return tb;
         }
 
         /// <summary>
-        /// Create a new <see cref="TermsBucket" from a given <see cref="DataRow"/>/>.
+        /// Create a new <see cref="RangeBucket" from a given <see cref="DataRow"/>/>.
         /// </summary>
         /// <param name="row">The row to be transformed to bucket.</param>
         /// <returns>A new TermsBucket.</returns>
-        public static TermsBucket CreateTermsBucketFromDataRow(DataRow row)
+        public static RangeBucket CreateRangeBucketFromDataRow(DataRow row)
         {
             Ensure.IsNotNull(row, nameof(row));
 
-            var key = row[(int)BucketColumnNames.Timestamp];
+            var range = Convert.ToString(row[(int)BucketColumnNames.SummarizeByColumn]);
             var count = row[BucketColumnNames.Count];
 
-            var tb = new TermsBucket
+            // Ignore the row for "other" records, that did not match the ranges
+            if (range == BucketColumnNames.RangeDefaultBucket)
+            {
+                return null;
+            }
+
+            // Parse the range
+            string[] splitRange = range.Split('-');
```

Comment:
I would do it like this:
```c#
var splitRange = range
                .Split('-')
                .Select(s => string.IsNullOrEmpty(s) ? (double?)null : double.Parse(s))
                .ToArray();
var from = splitRange[0];
var to = splitRange[1];
```
But if you prefer yours it's also fine

Reply:
I switched to your version